src/status_im/ui/screens/ens/views.cljs

```diff
@@ -0,0 +1,511 @@
+  [react/view {:style {:height           60
+                       :background-color colors/white
+                       :border-top-width 1
+                       :border-top-color colors/gray-lighter}}
+   [react/view {:style {:margin-horizontal 16
+                        :flex-direction    :row
+                        :justify-content   :space-between}}
+    [react/view {:flex-direction :row}
+     [react/view {:style {:margin-top 12 :margin-right 8}}
+      [components.common/logo
+       {:size      36
+        :icon-size 16}]]
+     [react/view {:flex-direction :column :margin-vertical 8}
+      [react/text {:style {:font-size 15}}
+       (i18n/label :t/ens-10-SNT)]
+      [react/text {:style {:color colors/gray :font-size 15}}
+       (i18n/label :t/ens-deposit)]]]
+    [button {:disabled?    (not @checked)
+             :label-style  (when (not @checked) {:color colors/gray})
+             :on-press     #(on-registration props)}
+     (i18n/label :t/ens-register)]]])
+
+(defn- registration [{:keys [address public-key] :as props}]
+  [react/view {:style {:flex 1 :margin-top 24}}
+   [section {:title   (i18n/label :t/ens-wallet-address)
+             :content address}]
+   [react/view {:style {:margin-top 14}}
+    [section {:title   (i18n/label :t/key)
+              :content public-key}]]
+   [agreement props]])
+
+(defn- icon [{:keys [state]}]
+  [react/view {:style {:margin-top 68 :margin-bottom 24  :width 60 :height 60 :border-radius 30
+                       :background-color colors/blue :align-items :center :justify-content :center}}
+   [vector-icons/icon (main-icon state) {:color colors/white}]])
+
+(defn- username-input [{:keys [custom-domain? username state] :as props} usernames]
+  [react/view {:flex-direction :row}
+   [react/text-input {:on-change-text    #(on-username-change custom-domain? %)
+                      :on-submit-editing #(on-registration props)
+                      :auto-capitalize   :none
+                      :auto-correct      false
+                      :default-value     username
+                      :auto-focus        true
+                      :placeholder       (default-name custom-domain?)
+                      :style             {:flex 1 (if (= state :registering) :padding-horizontal :padding-left) 48
+                                          :text-align :center :font-size 22}}]
+   [input-icon props usernames]])
+
+(defn- final-state-label [state]
+  (case state
+    :registered
+    (i18n/label :t/ens-registered-title)
+    :saved
+    (i18n/label :t/ens-saved-title)
+    :registration-failed
+    (i18n/label :t/ens-registration-failed-title)
+    ""))
+
+(defn- final-state-details [{:keys [state username]}]
+  (case state
+    :registered
+    [react/text {:style {:color colors/gray :font-size 14 :text-align :center}}
+     (i18n/label :t/ens-registered)]
+    :registration-failed
+    [react/text {:style {:color colors/gray :font-size 14 :text-align :center}}
```

Comment:
Duplication for `[:registered :registration-failed]` can be removed with (keyword (str "ens-" (name state)))